staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1369,23 +1370,37 @@ func hasServedCRDVersion(spec *apiextensionsv1.CustomResourceDefinitionSpec, ver
 // buildOpenAPIModelsForApply constructs openapi models from any validation schemas specified in the custom resource,
 // and merges it with the models defined in the static OpenAPI spec.
 // Returns nil models ifthe static spec is nil, or an error is encountered.
-func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensionsv1.CustomResourceDefinition) (*spec.Swagger, error) {
+func buildOpenAPIModelsForApply(staticOpenAPISpec map[string]*spec.Schema, crd *apiextensionsv1.CustomResourceDefinition) (*spec3.OpenAPI, error) {
```

Comment:
How do you feel about returning a full OpenAPI here when only the components part was provided as an input? I don't feel like it makes a lot of sense. Could we build the typeconverter directly in here maybe? Or return whatever the typeconverter needs?

Reply:
I had an initial implementation that did as you said, but changed to this to keep the diff readable for now. Could return `spec.Components.Schemas` instead?

Reply:
done